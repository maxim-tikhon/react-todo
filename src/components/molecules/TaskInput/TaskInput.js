import CheckBoxInput from '../../atoms/CheckBoxInput/CheckBoxInput';
import TextInput from '../../atoms/TextInput/TextInput';
import styles from './TaskInput.module.scss';

function TaskInput({ onEnterTask }) {
  return (
    <>
      <CheckBoxInput disabled />
      <TextInput
        className={styles.newTaskInput}
        placeholder="Please enter a new task and click Enter to save"
        onEnter={onEnterTask}
      />
    </>
  );
}

export default TaskInput;
