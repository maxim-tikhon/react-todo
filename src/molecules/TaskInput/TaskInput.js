import CheckBoxInput from "../../atoms/CheckBoxInput/CheckBoxInput";
import TextInput from "../../atoms/TextInput/TextInput";
import styles from "./TaskInput.module.scss";

const TaskInput = ({ onEnterTask }) => {
	return (
		<div>
			<CheckBoxInput />
			<TextInput
				className={styles.newTaskInput}
				placeholder="Please enter a new task and click Enter to save"
				onEnter={onEnterTask}
			/>
		</div>
	);
}

export default TaskInput;
