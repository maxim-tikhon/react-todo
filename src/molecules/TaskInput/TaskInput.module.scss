.newTaskInput {
  margin-left: 10px;
  width: calc(100% - 2rem);
}