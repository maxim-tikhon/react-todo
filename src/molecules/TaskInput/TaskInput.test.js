import renderer from 'react-test-renderer';
import TaskInput from './TaskInput';

describe('TaskInput', () => {
	it('renders correctly', () => {
		const output = renderer.create(<TaskInput />).toJSON();
		expect(output).toMatchSnapshot();
	});
});
