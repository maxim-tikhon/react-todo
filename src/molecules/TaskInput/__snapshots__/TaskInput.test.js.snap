// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInput renders correctly 1`] = `
<div>
  <input
    className="customCheckbox"
    disabled={true}
    onChange={[Function]}
    type="checkbox"
  />
  <input
    className="textInput newTaskInput"
    onKeyDown={[Function]}
    placeholder="Please enter a new task and click Enter to save"
    type="text"
  />
</div>
`;
